### ISSUE TITLE
Improve Feathers accessibility

### ISSUE THREAD

**Issue by janhohenheim** [bevyengine/bevy#21119]
## What problem does this solve or what need does it fill?

Feathers is currently hardcoded to font size 14. This is too low for my eyes. But it looks like the only way to up this is by hacking in some observer that changes the font settings of every widget spawned.

## What solution would you like?

Feathers should have some blessed way of programmatically setting the font size. Ideally, also allow customizing the font itself and the background colors for even better accessibility.

## What alternative(s) have you considered?

- Leave Bevy only for the able-bodied
- Leave this to third-party apps
- Punt until we have a dedicated preferences API
- Document the observer trick in an example and bless that as the canonical way

## Additional context

Ideally, I would like editors to have an easy way of implementing something like letting the user press `CTRL +` to increase the font size and `CTRL -` to decrease it again.

**Comment by viridia** [comment]
There are a couple of different solutions of varying generality:

* Having an explicit `Size` option for `ButtonProps`. This would only apply to buttons that had that specific size property.
* Having support for font size in feathers themes. This would only apply to apps using feathers and the setting would be controlled by theme designers (or anyone who can mutate the theme struct).
* Adding support for relative font sizing generally. This could be used by any Bevy app that uses fonts.

The last approach is based on @ickshonpe 's work on font sizing: #9525 . Specifically, you would have a global resource, which I call `BaseFontSize(f32)` which establishes a global setting for font sizes. Then the text font size component would be changed to accept a `FontSize` enum, which has `Px`, `Vw` and `Base` variants. This is similar to `Val` but includes special options for text (much like CSS does).

So for example, a font with `FontSize::Base(1.5)` would have a point size 50% larger than the global base font.

Changing the `BaseFontSize` would cause an immediate increase in font sizes that used the `Base` variant. This might cause repositioning of some items but wouldn't increase the size of borders, padding, or other metrics like changing the UiScale would do.

In some cases, this might cause fonts to outgrow their containers - for example, the Feathers buttons have a fixed height of 24px (so that they can align with other Feathers widgets), if the font size increases beyond that then the font will extend outside of the button bounds.

**Comment by janhohenheim** [comment]
I think VS Code is pretty good in how they handle it, as a reference:

https://github.com/user-attachments/assets/93284f7d-c877-44b9-9020-6972f8865142

Note that their text doesn't escape the buttons no matter how big you make everything. Could we do that too?

**Comment by Freyja-moth** [comment]
If we are trying to improve the accessibility of feathers we should also consider allowing users to select their own fonts, and potentially including some accessible fonts by default.
